src/hotspot/share/opto/phasetype.hpp

```diff
@@ -138,13 +138,13 @@ enum CompilerPhaseType {
 };
 #undef table_entry
 
-static const char* phase_descriptions[] = {
+static constexpr const char* compiler_phase_descriptions[] = {
```

Comment:
A simpler and better solution would be to make `phase_descriptions` and `phase_names` static
data members of `CompilerPhaseTypeHelper`, with just a declaration in the header, and the
definition in a new .cpp file. (Note that with C++17 they could be declared `inline` and the .cpp
file isn't needed.)
I'm not sure why the change from `const` to `constexpr` is being made here.
Doesn't this have a problem that each translation unit including this header gets it's own
private copy of these arrays? And doesn't that introduce an ODR violation for the referring
code in CompilerPhaseTypeHelper? (Maybe the `constexpr` change has something to
do with that? But I'm not sure how.)